dashboard/app/controllers/api/v1/regional_partners_controller.rb

```diff
@@ -77,4 +92,40 @@ def get_partner_cohort_capacity(regional_partner_value, role)
     end
     nil
   end
+
+  # Get the regional partner's number of enrollments to current-year role-specific workshops.
+  #
+  # @param regional_partner_id [Integer]
+  # @param role [String] a string contains workshop course and subject type, e.g. 'csp_teachers'
+  # @return [Integer, nil] nil if regional_partner_id is invalid, otherwise returns a count
+  # @see Api::V1::Pd::ApplicationsController:ROLES
+  #
+  def get_partner_enrollment_count(regional_partner_id, role)
+    partner = RegionalPartner.find_by_id(regional_partner_id)
+    return unless partner
+
+    workshops = partner.pd_workshops
+    workshops_by_role =
+      case role.to_sym
+      when :csf_facilitators
+        workshops.where(course: COURSE_CSF, subject: SUBJECT_CSF_FIT)
+      when :csd_facilitators
+        workshops.where(course: COURSE_CSD, subject: SUBJECT_CSD_FIT)
+      when :csp_facilitators
+        workshops.where(course: COURSE_CSP, subject: SUBJECT_CSP_FIT)
+      when :csd_teachers
+        workshops.where(course: COURSE_CSD).where.not(subject: SUBJECT_CSD_FIT)
+      when :csp_teachers
+        workshops.where(course: COURSE_CSP).where.not(subject: SUBJECT_CSP_FIT)
+      else
+        # Should never get here. This value will cause an exception on the next calculation,
+        # which expects an ActiveRecord::Relation.
+        nil
+      end
+
+    workshop_current_year_ids =
+      workshops_by_role.select {|ws| ws.school_year == APPLICATION_CURRENT_YEAR}.pluck(:id)
```

Comment:
Since `workshops_by_role` is still an `ActiveRecord::Relation` I think the `select` will load the ActiveRecord models, and we could avoid that with an additional `where` clause instead:

```suggestion
      workshops_by_role.where(school_year: APPLICATION_CURRENT_YEAR).pluck(:id)
```

Reply:
Seems like ActiveRecord doesn't allow function in where clause. `school_year` is a function in `workshop.rb` file (which is also in this PR). 

However, the original query could be problematic because it executes 1 more query for each workshop. So the maximum number of queries is equal to number of workshops (of a certain type) a regional partner has.

Reply:
It take roughly 1-2 seconds to check 116 workshops in production console. I think we can live with this for now. The optimization is not trivial. We can come back to it later if we see trouble.

Reply:
Ah, I missed that `school_year` was a function, not a column.  Looks good!